exist-core/src/main/java/org/exist/xquery/RangeSequence.java

```diff
@@ -208,6 +210,25 @@ public MemoryNodeSet toMemNodeSet() throws XPathException {
     public void removeDuplicates() {
     }
 
+    @Override
+    public boolean containsReference(final Item item) {
+        return start == item || end == item;
+    }
+
+    @Override
+    public boolean contains(final Item item) {
+        if (item instanceof IntegerValue) {
+            try {
+                final BigInteger other = item.toJavaObject(BigInteger.class);
+                return other.compareTo(start.toJavaObject(BigInteger.class)) >= 0
+                        && other.compareTo(end.toJavaObject(BigInteger.class)) <= 0;
+            } catch (final XPathException e) {
+                return false;
```

Comment:
Should we not log the error here somewhere?

Reply:
Okay fixed.